src/scheduler/pbs_sched.c

```diff
@@ -642,29 +646,33 @@ connect_server(char *svrhost)
 		}
 	}
 	if (tpp_em_add_fd(poll_context, secondary_sock, EM_IN | EM_HUP | EM_ERR) < 0) {
-		log_errf(-1, __func__, "Couldn't add secondary connection to poll list for server %s", svrhost);
+		log_errf(errno, __func__, "Couldn't add secondary connection to poll list for server %s", svrhost);
 		pbs_disconnect(primary_sock);
 		pbs_disconnect(secondary_sock);
 		die(-1);
 	}
 
 	svr = malloc(sizeof(sched_svrconn));
 	if (!svr) {
-		log_errf(-1, __func__, MEM_ERR_MSG);
+		log_err(errno, __func__, MEM_ERR_MSG);
 		die(-1);
 	}
 
 	svr->svrhost = strdup(svrhost);
 	if (!svr->svrhost) {
 		free(svr);
-		log_errf(-1, __func__, MEM_ERR_MSG);
+		log_err(errno, __func__, MEM_ERR_MSG);
 		die(-1);
 	}
 	svr->primary_sock = primary_sock;
 	svr->secondary_sock = secondary_sock;
-	i = 0;
-	while (servers[i] != NULL) i++;
-	servers[i] = svr;
+	svrs_newarr = add_ptr_to_array(servers, svr);
+	if (svrs_newarr == NULL) {
+		close_server(svr);
+		/* no need of logging as add_ptr_to_array already logged error */
```

Comment:
It is standard practice in the scheduler to log at the point of the malloc().  There is no need for this comment.